pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -405,6 +408,48 @@ func (configgen *ConfigGeneratorImpl) buildGatewayInboundHTTPRouteConfig(
 	return out
 }
 
+// mergeIngress will append rules from Ingress-generated virtual services to existing virtual services.
+// The ingress can be created dynamically, and only has very limitted capabilities -
+// [host]/path->backend configuration
+func mergeIngress(configs []model.Config) []model.Config {
+	configsOut := []model.Config{}
+	hostsToVS := map[string]*networking.VirtualService{}
+
+	// Ignore ingress resources, build a map of hosts to (real) VirtualServices.
+	for _, c := range configs {
+		vs := c.Spec.(*networking.VirtualService)
+		for _, h := range vs.Hosts {
+			if !strings.HasSuffix(c.Name, model.IstioIngressGatewayName) {
+				vse, e := hostsToVS[h]
+				if e {
+					log.Warnf("Duplicated hostname %s OLD:%v NEW:%v", h, vse, vs)
```

Comment:
?? Why are you doing this for non ingress stuff?

Reply:
See above, will move to separate PR or file a bug. We probably need to do same trick Nate used to use the timestamp to reject most recent virtual service in cases of duplicated hostname.